### PR TITLE
fixes PEN-14

### PR DESCRIPTION
**Pull request by patrick330602**
Already tested locally and it works for me. Need more tests and it is not the final script.

### CODE REVIEW

File: profile.d/00-pengwin.sh [hunk 1]
```diff
@@ -1,30 +1,72 @@
...
-  wsl2_d_tmp="$(eval "$ipconfig_exec" | grep -n -m 1 "Default Gateway.*: [0-9a-z]" | cut -d : -f 1)"
-  if [[ ${wsl2_d_tmp} ]]; then
-    wsl2_d_tmp="$(eval "$ipconfig_exec" | sed ''"$(expr $wsl2_d_tmp - 4)"','"$(expr $wsl2_d_tmp + 0)"'!d' | grep IPv4 | cut -d : -f 2 | sed -e "s|\s||g" -e "s|\r||g")"
+  if ( eval "$ipconfig_exec" | grep -n -m 1 "Default Gateway.*: [0-9a-z]" | cut -d : -f 1 ) >/dev/null; then
+    wsl2_d_tmp="$(eval "$ipconfig_exec" | grep -n -m 1 "Default Gateway.*: [0-9a-z]" | cut -d : -f 1)"
+    wsl2_d_tmp="$(eval "$ipconfig_exec" | sed $(expr $wsl2_d_tmp - 4)','$(expr $wsl2_d_tmp + 0)'!d' | grep IPv4 | cut -d : -f 2 | sed -e "s|\s||g" -e "s|\r||g")"
     export DISPLAY=${wsl2_d_tmp}:0.0
+
+    # check if we have wsl.exe in path
+    if ( which wsl.exe >/dev/null ); then
```
[the remark this answers is not in the document]
Comment (patrick330602) [reply profile.d/00-pengwin.sh]: I will try to complete these requests today

**Comment by patrick330602** [comment]
I have updated the PR according to the suggestions

File: profile.d/00-pengwin.sh [hunk 2]
```diff
@@ -1,30 +1,59 @@
 
 # check whether it is WSL1 for WSL2
+
+if [ ! -f /etc/sudoers.d/pengwin-wsl-profile ]; then
+  echo "First time configurating Pengwin; password required"
+  (echo "%sudo   ALL=NOPASSWD: /usr/local/bin/wsl_change_checker" | sudo EDITOR='tee' visudo --quiet --file=/etc/sudoers.d/pengwin-wsl-profile) &>/dev/null
```
[the remark this answers is not in the document]
Comment (patrick330602) [reply profile.d/00-pengwin.sh]: Yeah, you are right, that would work better

File: bin/wsl_change_checker [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+source /etc/environment
+
+wsl2_type=$1; shift
+wsl2_readable_type=$1; shift
+wsl_display_cont=$1
+
+if ( which wsl.exe >/dev/null ); then
+  if [ ! $WSL2 -eq $wsl2_type ]; then
```
[the remark this answers is not in the document]
Comment (patrick330602) [reply bin/wsl_change_checker]: Added another cli for that

File: bin/wsl2_ip_checker [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+source /etc/environment
+
+passed_new_ip_address="$1"; shift
+
+env_wsl2_ip_address="$(grep -e ^DISPLAY=.* /etc/environment | sed -e 's/DISPLAY=\(.*\):0.*$/\1/g')"
+
+  if [ ! "$passed_new_ip_address" -eq "$env_wsl2_ip_address" ]; then
```
Comment (crramirez) [inline bin/wsl2_ip_checker]: un indent

**Pull request merged**